sixtyfps_compiler/lib.rs

```diff
@@ -122,6 +122,21 @@ pub async fn compile_syntax_node(
 
     let mut loader =
         typeloader::TypeLoader::new(global_type_registry, &compiler_config, &mut diagnostics);
+
+    let known_styles = fileaccess::styles();
+    let current_style = loader.style();
+    if !known_styles.contains(&current_style.as_ref()) {
+        diagnostics.push_diagnostic_with_span(
+            format!(
+                "Style \"{}\" is unknown. Use one of [{}] instead",
+                &current_style,
+                known_styles.join(", ")
+            ),
+            Default::default(),
+            crate::diagnostics::DiagnosticLevel::Error,
+        );
+    }
+
     let foreign_imports =
         loader.load_dependencies_recursively(&doc_node, &mut diagnostics, &type_registry).await;
```

Comment:
We should probably treat wrong style as a fatal error, and do not process further.